rust/src/writer/mod.rs

```diff
@@ -16,15 +16,15 @@ pub use record_batch::RecordBatchWriter;
 
 pub mod json;
 pub mod record_batch;
-pub(crate) mod stats;
+pub mod stats;
 pub mod utils;
 
 #[cfg(test)]
 pub mod test_utils;
 
 /// Enum representing an error when calling [`DeltaWriter`].
 #[derive(thiserror::Error, Debug)]
-pub(crate) enum DeltaWriterError {
+pub enum DeltaWriterError {
```

Comment:
Not exposing too many error variants is actually deliberate :). In the end we want to get to a place where we only expose `DeltaError` to the outside.

Reply:
I understand. Should I return DeltaTableError instead of DeltaWriterError in create_add or do you suggest something else?